portal-api/src/v1/api.js

```diff
@@ -471,6 +472,45 @@ const getPortalFacilityAmendment = async (facilityId, amendmentId) => {
   }
 };
 
+/**
+ * Upserts a draft amendment for a portal facility in the database.
+ * @param {Object} params
+ * @param {string} params.dealId - id of the deal with the relevant facility.
+ * @param {string} params.facilityId - id of the facility to amend.
+ * @param {import('@ukef/dtfs2-common').PortalFacilityAmendmentUserValues} params.amendment - the draft amendment to be upserted.
+ * @param {import('@ukef/dtfs2-common').AuditDetails} params.auditDetails - The audit details for the update.
+ * @returns {Promise<(import('@ukef/dtfs2-common').PortalFacilityAmendmentWithUkefId)>} - the amendment
+ */
+const putPortalFacilityAmendment = async ({ dealId, facilityId, amendment, auditDetails }) => {
+  try {
+    const response = await axios({
+      method: 'put',
+      url: `${DTFS_CENTRAL_API_URL}/v1/portal/facilities/${facilityId}/amendments`,
+      headers: headers.central,
+      data: {
+        dealId,
+        facilityId,
+        amendment,
+        auditDetails,
+      },
+    });
+
+    const { success, error, data } = PORTAL_FACILITY_AMENDMENT_WITH_UKEF_ID.safeParse(response.data);
+
+    if (success) {
+      return data;
+    }
+
+    const formattedErrors = getFormattedZodErrors(error);
+    console.error('Type validation error occurred when receiving portal amendment from dtfs-central:', formattedErrors);
```

Comment:
```suggestion
    console.error('Type validation error occurred when receiving portal amendment from dtfs-central %o', error);
```

Reply:
Trying to move away from using `getFormattedZodErrors` and `getErrorCode`. Thanks

Reply:
Changed, thanks